k9mail/src/main/java/com/fsck/k9/mailstore/LocalFolder.java

```diff
@@ -892,12 +894,9 @@ public LocalMessage doDbWork(final SQLiteDatabase db) throws WrappedException, U
         }
     }
 
-    public List<LocalMessage> getMessages(String[] uids, MessageRetrievalListener<LocalMessage> listener)
+    public List<LocalMessage> getMessagesByUids(@NonNull  String[] uids)
```

Comment:
It looks like this message is only called once and the caller has to convert a list to an array just to call this method. However, this would work just fine with a `List`.